server/src/main/java/server/haengdong/application/ImageService.java

```diff
@@ -30,18 +32,26 @@ public class ImageService {
     private String directoryPath;
 
     private final S3Client s3Client;
+    private final ExecutorService executorService;
 
     public List<String> uploadImages(List<MultipartFile> images) {
-        return images.stream()
-                .map(this::uploadImage)
+        List<CompletableFuture<String>> futures = images.stream()
+                .map(image -> CompletableFuture.supplyAsync(() -> uploadImage(image), executorService))
                 .toList();
+
+        CompletableFuture<List<String>> result = CompletableFuture.allOf(futures.toArray(new CompletableFuture[futures.size()]))
```

Comment:
```suggestion
        CompletableFuture<List<String>> result = CompletableFuture.allOf(new CompletableFuture[0]))
```
<img width="1264" alt="image" src="https://github.com/user-attachments/assets/1800a82a-6237-4dc8-b9fc-c8898910e58f">

futures 리스트의 정확한 크기가 필요한 것이 아니라, 명시적인 타입을 제공하고 배열 리사이징 최적화를 이용할 수 있도록 `new CompletableFuture[0]`를 매개변수로 넘겨주는게 더 좋겠네요 👍

Reply:
@Arachneee 확인 후 자체 머지 해주세요.